src/validator.rs

```diff
@@ -1,11 +1,11 @@
 use crate::{MpesaError, MpesaResult};
 
 pub trait PhoneValidator {
-    fn validate(&self) -> MpesaResult<()>;
+    fn validate_number(&self) -> MpesaResult<()>;
```

Comment:
I think the `PhoneValidator` holds the context of what is getting validated, therefore, the method can remain as `validate`

Perhaps the trait can be renamed to `PhoneNumberValidator`